### ISSUE TITLE
Migrate JavaLoggingSystemTests and Log4J2LoggingSystemTests to JUnit 5

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#17097]
Currently ignoring from the checkstyle rule since they need to be migrated

**Comment by L00kian** [comment]
It seems that having 
`@RunWith(ModifiedClassPathRunner.class) @ClassPathExclusions("log4j-*.jar")`
has no effect on LogbackLoggingSystemTests outcome. Should it be migrated as part of this ticket as well?

**Comment by philwebb** [comment]
Thanks for looking at this @L00kian. I should have marked this as blocked since we haven't get got a fix for #16957 since we need some upstream JUnit 5 changes.

**Comment by L00kian** [comment]
Is there any specific reason why modified class path excluding log4j needs to be on logback test? For what I have checked - the annotation presence has no effect on the test results. I can raise a separate PR with the logback tests migrated to Junit 5 along with the Log4j and Java logging ones

**Comment by wilkinsona** [comment]
I added it in [this commit](https://github.com/spring-projects/spring-boot/commit/cafe6dbbb1f58b21c37fbff169e11fdc8db67653). Unfortunately, I can't remember why I did so and didn't note the reason in the commit message. I suspect it was to do with not wanting the test to rely on class path ordering. However, if it works without the modified class path, it would be a welcome reduction in complexity to get rid of it.
